src/main/java/com/google/sps/servlets/TaskServlet.java

```diff
@@ -0,0 +1,144 @@
+        if (!taskDetail.equals("")) {
+            long timestamp = System.currentTimeMillis();
+
+            // Create an Entity that stores the input comment
+            Entity taskEntity = new Entity("Task");
+            taskEntity.setProperty("detail", taskDetail);
+            taskEntity.setProperty("timestamp", timestamp);
+            taskEntity.setProperty("reward", rewardPts);
+            taskEntity.setProperty("status", "OPEN");
+            taskEntity.setProperty("Owner", "Leonard");
+            taskEntity.setProperty("Helper", "N/A");
+            taskEntity.setProperty("Address", "4xxx Cxxxxx Avenue, Pittsburgh, PA 15xxx");
+
+            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+            datastore.put(taskEntity);
+        }
+
+        // Redirect back to the user page.
+        response.sendRedirect("/user_profile.html");
+    }
+
+    /** Return the input rewarding points by the user, or -1 if the input was invalid */
+    private int getRewardingPoints(HttpServletRequest request, String inputName) {
+        // Get the input from the form.
+        String rewardPtsString = request.getParameter(inputName);
+
+        // Convert the input to an int.
+        int rewardPts;
+        try {
+            rewardPts = Integer.parseInt(rewardPtsString);
+        } catch (NumberFormatException e) {
+            System.err.println("Could not convert to int: " + rewardPtsString);
+            return -1;
+        }
+
+        // Check that the input is positive.
+        if (rewardPts < 0 || rewardPts > 200) {
+            System.err.println("User input is out of range: " + rewardPtsString);
+            return -1;
+        }
+
+        return rewardPts;
+    }
+
+    @Override
+    public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String keyString = request.getParameter("key");
+
+        Key taskKey = KeyFactory.stringToKey(keyString);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.delete(taskKey);
+
+        // Redirect to the user profile page
+        response.sendRedirect("/user_profile.html");
+    }
+
+    @Override
+    public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String keyString = request.getParameter("key");
+        int rewardPts = getRewardingPoints(request, "edit-input");
+        if (rewardPts == -1) {
+            response.setContentType("text/html");
+            response.getWriter().println("Please enter a valid integer in the range of 0-200");
+            return;
+        }
+
+        // Get the task detail from the form input
+        String taskDetail = "";
+        String input = request.getParameter("edit-content-input");
+        if (input != null) {
+            taskDetail = input;
+        }
+
+        // If the input is nonempty and valid, set the taskDetail value to the input value
```

Comment:
This is still pending, right? I wasn't sure if you meant to leave it unfinished for now or if you accidentally deleted some lines here. It might be good in the meantime to redirect users back to the userpage, right now it sends you to an empty blank page.

Reply:
Yeah I am still working on the doPut method, but I will fill this out in the next commit.